sphinx/testing/fixtures.py

```diff
@@ -228,4 +228,6 @@ def tempdir(tmpdir):
     temporary directory that wrapped with `path` class.
     this fixture is for compat with old test implementation.
     """
+    # TODO(stephenfin): Deprecate this in favour of pytest's native tmpdir
+    # extension: https://docs.pytest.org/en/latest/tmpdir.html
```

Comment:
Hmm? Please tell me why. Sphinx's `tempdir` is a bit useful to me.

Reply:
Because of this line

> this fixture is for compat with old test implementation.

I found it rather difficult to use, in that you need to configure an environment variable to use it and if you don't, it defaults to using '/tmp' instead of a subdirectory in that folder like I thought it would. I will address the TODO in a future change, I assure you 😄

Reply:
Hmm? This is a simple wrapper for `tmpdir` fixture of pytest's. It returns path object instead string.

With using this, we can create a text file like following:
```
def test(tempdir):
    (tempdir / "filename.txt").write_text("hello sphinx!")
```

I believe this is almost same as `tmpdir` fixture. What do you want to go?

Reply:
Maybe I've misunderstood something here so. What I want to remove is the `_initialize_test_directory` function in `tests/conftest.py`. IMO, by using the `tempdir` fixture, we should automatically get a subdirectory in `TMPDIR` that's used for the entire test. This doesn't seem to be the case. Maybe I've just misunderstood something though. I've dropped this TODO and can look at this in more detail later.